### PR TITLE
changing loop for C++11 style

### PR DESCRIPTION
**Pull request by Joilnen**
to review

### CODE REVIEW

File: OgreMain/src/OgreRenderSystem.cpp [hunk 1]
```diff
@@ -618,10 +618,9 @@ namespace Ogre {
     void RenderSystem::shutdown(void)
     {
         // Remove occlusion queries
-        for (HardwareOcclusionQueryList::iterator i = mHwOcclusionQueries.begin();
-            i != mHwOcclusionQueries.end(); ++i)
+        for (auto &a : mHwOcclusionQueries)
```
Comment (paroj) [inline OgreMain/src/OgreRenderSystem.cpp:621]:
```suggestion
        for (auto q : mHwOcclusionQueries)
```
as this a pointer, `auto&` does not save anything. Also `q` for query is more descriptive as a running variable

**Review by Joilnen — COMMENTED** [review]
to review

File: OgreMain/src/OgreRenderSystem.cpp [hunk 2]
```diff
@@ -240,28 +240,20 @@ namespace Ogre {
     //-----------------------------------------------------------------------
     void RenderSystem::_initRenderTargets(void)
     {
-
-        // Init stats
-        for(
-            RenderTargetMap::iterator it = mRenderTargets.begin();
-            it != mRenderTargets.end();
-            ++it )
+        for(auto &&a : mRenderTargets)
```
Comment (paroj) [inline OgreMain/src/OgreRenderSystem.cpp:243]: please use `auto& p` when iterating maps.
- The name (pair) should correspond to the element type.
- rvalue at this place is confusing and has no advantage

File: OgreMain/src/OgreRenderSystem.cpp (hunk 1 above)
Comment (Joilnen) [reply OgreMain/src/OgreRenderSystem.cpp:611]: but in this case it makes copy, using reference wouldn't be more efficient ?

Comment (paroj) [reply OgreMain/src/OgreRenderSystem.cpp:611]: no, a reference has the same cost as copying a pointer. Sometimes more as a copy tells the compiler that the original cannot change.

**Comment by Joilnen** [comment]
I will resubmit

**Comment by paroj** [comment]
you can also force-push to an existing branch, if that is easier for you than opening new PRs

**Pull request closed without merging**